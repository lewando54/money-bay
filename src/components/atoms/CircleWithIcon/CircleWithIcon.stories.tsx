/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import CircleWithIcon from './CircleWithIcon'

const meta: Meta<typeof CircleWithIcon> = {
    component: CircleWithIcon,
    argTypes: {
        backgroundColor: { control: 'object' },
        castShadow: { control: 'boolean' },
        onClick: {action: 'clicked', disable: true }
    }
}

export default meta
type Story = StoryObj<typeof CircleWithIcon>

export const Default: Story = {
    args: {
        backgroundColor: '#fff',
        onClick: null
    }
}

export const Pressable: Story = {}

export const CustomBackgroundColor: Story = {
    args: {
        backgroundColor: '#F00',
        onClick: null
    }
}

export const Gradient: Story = {
    args: {
        backgroundColor: ['#6B99CB', '#161966'],
        onClick: null
    }
}

export const CastShadow: Story = {
    args: {
        backgroundColor: '#fff',
        castShadow: true,
        onClick: null
    }
}
