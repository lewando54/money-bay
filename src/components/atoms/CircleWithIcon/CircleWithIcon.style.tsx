import { StyleSheet } from 'react-native'

interface ICircleWithIconStyleProps {
    size: number
    backgroundColor: string
}

export default ({size, backgroundColor}: ICircleWithIconStyleProps) => StyleSheet.create({
    circle: {
        width: size,
        height: size,
        borderRadius: size * 2,
        backgroundColor: backgroundColor,
        justifyContent: 'center',
        alignItems: 'center',
    }
})
