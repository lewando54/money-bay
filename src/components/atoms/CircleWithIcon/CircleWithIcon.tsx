import { TouchableOpacity, Image, View } from 'react-native'
import circleWithIconStyle from './CircleWithIcon.style'
import { AntDesign } from '@expo/vector-icons'

interface ICircleWithIconProps {
    size?: number,
    icon?: React.ReactNode,
    backgroundColor?: string,
    onClick?: () => void
    testID?: string
}

export default function CircleWithIcon({
    size=40,
    icon=<AntDesign size={size} name={'question'} color={'#55ACEE'}/>,
    backgroundColor='#F1F5FD',
    onClick,
    testID
}: ICircleWithIconProps){
    if(onClick){
        return (
            <TouchableOpacity 
                style={[
                    circleWithIconStyle({size, backgroundColor}).circle
                ]}
                testID={testID}
                onPress={onClick}
            >
                {icon}
            </TouchableOpacity>
        )
    }
    else{
        return (
            <View 
                style={[
                    circleWithIconStyle({size, backgroundColor}).circle
                ]}
                testID={testID}
            >
                {icon}
            </View>
        )
    }
}
