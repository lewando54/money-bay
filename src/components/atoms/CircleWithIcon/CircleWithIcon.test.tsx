import React from 'react'
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react-native'

import CircleWithIcon from './CircleWithIcon'
describe('CircleWithIcon', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<CircleWithIcon />)
        expect(tree).toMatchSnapshot()
    })

    it('should call onClick function when clicked', () => {
        const onClick = jest.fn()
        const { getByTestId } = render(<CircleWithIcon testID='test' onClick={onClick}/>)
        fireEvent.press(getByTestId('test'))
        expect(onClick).toHaveBeenCalled()
    })

    it('should be gradient and call onClick when passed array of colors', () => {
        const onClick = jest.fn()
        const { getByTestId } = render(<CircleWithIcon testID='test' onClick={onClick} backgroundColor={['#f00', '#ff0']}/>)
        fireEvent.press(getByTestId('test'))
        expect(onClick).toHaveBeenCalled()
    })

    it('should be gradient when passed array of colors', () => {
        const tree = renderer.create(<CircleWithIcon backgroundColor={['#f00', '#ff0']}/>)
        expect(tree).toMatchSnapshot()
    })

    it('should be gradient when passed array of colors and positioned with start/end', () => {
        const tree = renderer.create(<CircleWithIcon backgroundColor={['#f00', '#ff0']} start={{x: 0, y: 0}} end={{x: 1, y: 1}}/>)
        expect(tree).toMatchSnapshot()
    })

    it('should render with shadow', () => {
        const tree = renderer.create(<CircleWithIcon castShadow={true} />)
        expect(tree).toMatchSnapshot()
    })

    it('should render with shadow when clickable', () => {
        const onClick = jest.fn()
        const { getByTestId } = render(<CircleWithIcon castShadow={true} testID='test' onClick={onClick}/>)
        fireEvent.press(getByTestId('test'))
        expect(onClick).toHaveBeenCalled()
    })

    it('should render with shadow when clickable and gradient is applied', () => {
        const onClick = jest.fn()
        const { getByTestId } = render(<CircleWithIcon castShadow={true} testID='test' onClick={onClick} backgroundColor={['#f00', '#ff0']}/>)
        fireEvent.press(getByTestId('test'))
        expect(onClick).toHaveBeenCalled()
    })

    it('should render with shadow when gradient is applied', () => {
        const tree = renderer.create(<CircleWithIcon castShadow={true} backgroundColor={['#f00', '#ff0']}/>)
        expect(tree).toMatchSnapshot()
    })
})