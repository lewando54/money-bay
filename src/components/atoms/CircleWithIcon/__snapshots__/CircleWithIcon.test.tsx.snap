// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleWithIcon should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#F1F5FD",
        "borderRadius": 80,
        "height": 40,
        "justifyContent": "center",
        "width": 40,
      },
    ]
  }
>
  <Text />
</View>
`;
